src/utilities/alfalfa/AlfalfaComponent.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef ALFALFA_COMPONENT_HPP
+#define ALFALFA_COMPONENT_HPP
+
+#include <json/json.h>
+
+#include "../UtilitiesAPI.hpp"
+
+namespace openstudio {
+namespace alfalfa {
+  enum Capability
+  {
+    Input = 1,
+    Output = 2
+  };
+
+  inline Capability operator|(const Capability& a, const Capability& b) {
+    return static_cast<Capability>(static_cast<int>(a) | static_cast<int>(b));
+  }
+
+  inline Capability operator&(const Capability& a, const Capability& b) {
+    return static_cast<Capability>(static_cast<int>(a) & static_cast<int>(b));
+  }
+
+  class UTILITIES_API AlfalfaComponent
+  {
+   public:
+    AlfalfaComponent(const std::string& type, const Capability capabilities);
+
+    /**
+     * Get if the component can be used as an Input to the model
+     */
+    bool canInput() const;
+
+    /**
+     * Get if the component can be used as an Output from the model
+     */
+    bool canOutput() const;
+
+    bool operator==(const AlfalfaComponent& rhs) const;
+
+    /**
+     * A string which reflects the type of the component
+     */
+    std::string type;
+
```

Comment:
Enum class instead of string? Seems like this is a limited number of valid things.

Reply:
Yeah, I also didn't like this. Usually it would all be type inference so you wouldn't have to maintain in multiple places. Depending on where we decide to go with the component it could go one of a couple different ways.